pkg/applicationmanager/controllers/applicationrestore.go

```diff
@@ -163,13 +163,81 @@ func (a *ApplicationRestoreController) verifyNamespaces(restore *storkapi.Applic
 func (a *ApplicationRestoreController) createNamespaces(backup *storkapi.ApplicationBackup,
 	backupLocation string,
 	restore *storkapi.ApplicationRestore) error {
+	logrus.Info("Starting createNamespaces")
 	var namespaces []*v1.Namespace
 
+	// rtTemplate := &storkapi.ResourceTransformation{}
+	// a.client.Get(context.Background(), types.NamespacedName{Name: restore.Spec.ResourceTransformationTemplate, Namespace: kdmputils.AdminNamespace}, rtTemplate)
+
+	var gvkResourceTransform map[schema.GroupVersionKind][]storkapi.TransformSpecs
+	if restore.Spec.ResourceTransformationTemplate != "" {
+		var err error
+		gvkResourceTransform, err = resourcecollector.GetGVKToTransformSpecMapper(restore.Spec.ResourceTransformationTemplate, kdmputils.AdminNamespace)
+		if err != nil {
+			log.ApplicationRestoreLog(restore).
+				Warnf("Unable to get transformation spec from :%s, skipping transformation for this restore, err: %v", restore.Spec.ResourceTransformationTemplate, err)
+			return err
+		}
+	}
+
+	logrus.Info("gvkResourceTransform: ", gvkResourceTransform)
+
 	nsData, err := a.downloadObject(backup, backupLocation, restore.Namespace, nsObjectName, true)
 	if err != nil {
 		return err
 	}
 
+	logrus.Info("Before nsdata: ", string(nsData))
+
+	if gvkResourceTransform != nil {
+		objects := make([]*unstructured.Unstructured, 0)
+		if err = json.Unmarshal(nsData, &namespaces); err != nil {
```

Comment:
This sound like a hack because of the way we store namespace resources., We should have fixed while we are backing up namespaces with GVK info in namespace.json. While doing actual implementation lets see if we have to chnage in backup path and how big that challenge would be

Reply:
TODO for the actual implementation